### PR TITLE
address CORE-3015

### PR DESCRIPTION
**Pull request by mmaxim**
@maxtaco I moved a bunch of stuff around, and also started properly responding to `RevokeSessionIDs`

### CODE REVIEW

File: rpc/server/session_cacher.go [hunk 1]
```diff
@@ -43,6 +45,24 @@ func NewSessionCacher(a gregor1.AuthInterface, cl clockwork.Clock, timeout time.
 	return sc
 }
 
+func NewSessionCacherURI(ss *rpc.FMPURI, cl clockwork.Clock, timeout time.Duration,
```
Comment (maxtaco) [inline rpc/server/session_cacher.go]: Rename to: `NewSessionCacheFromURI` ?

File: rpc/server/session_cacher.go [hunk 2]
```diff
@@ -94,6 +110,12 @@ func (sc *SessionCacher) clearExpiryQueue() {
 	}
 }
 
+func (sc *SessionCacher) reconnectAuthdHandler(handler authdHandler) {
+	for cli := range handler.connectCh {
+		sc.parent = gregor1.AuthClient{cli}
```
Comment (maxtaco) [inline rpc/server/session_cacher.go]: this might not be go-routine safe. I'm not quite sure...

Comment (maxtaco) [inline rpc/server/session_cacher.go]: IS there a chance one goroutine will be setting this pointer while another will be acccessing it?

**Comment by maxtaco** [comment]
a few comments

File: rpc/server/session_cacher.go (hunk 1 above)
Comment (mmaxim) [inline rpc/server/session_cacher.go]: Yep!

**Comment by mmaxim** [comment]
udpated!

**Comment by maxtaco** [comment]
This looks great. Looks ready to go, but could maybe benefit from a few comments in the trickiest spots.

**Comment by maxtaco** [comment]
👍 but add a few comments.

**Pull request merged**